kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/arbitrary/ReflectiveBindTest.kt

```diff
@@ -123,6 +129,55 @@ class ReflectiveBindTest : StringSpec(
          )
       }
 
+      "Can bind to all types that can be used as parameters" {
+         val enumArb = Arb.bind<Shape>()
+         enumArb.next().shouldBeInstanceOf<Shape>()
+
+         val listArb = Arb.bind<List<Int>>()
+         listArb.next().shouldBeInstanceOf<List<Int>>()
+
+         val bigDecimalArb = Arb.bind<BigDecimal>()
+         bigDecimalArb.next().shouldBeInstanceOf<BigDecimal>()
+      }
+
+      "Can bind to no-arg constructor classes" {
+         val noArgArb = Arb.bind<NoArgConstructor>()
+         noArgArb.next().shouldBeInstanceOf<NoArgConstructor>()
+      }
+
+      "Can bind to sealed classes" {
+         val shape3dArb = Arb.bind<Shape3d>()
+         shape3dArb.next().shouldBeInstanceOf<Shape3d>()
+
+         val shapes3d = shape3dArb.take(100).toList()
```

Comment:
Do you want to specify a random seed to ensure this test and all others which uses this pattern not flakey?